src/linker.c

```diff
@@ -420,13 +420,13 @@ int load_program(jq_state *jq, struct locfile* src, block *out_block) {
     return 1;
   }
 
-  char* home = getenv("HOME");
-  if (home) {    // silently ignore no $HOME
-    /* Import ~/.jq as a library named "" found in $HOME */
+  jv home = get_home();
+  if (jv_is_valid(home)) {    // silently ignore if home dir not determined
+    /* Import ~/.jq as a library named "" found in $HOME or %USERPROFILE% */
     block import = gen_import_meta(gen_import("", NULL, 0),
         gen_const(JV_OBJECT(
             jv_string("optional"), jv_true(),
-            jv_string("search"), jv_string(home))));
+            jv_string("search"), home)));
     program = BLOCK(import, program);
   }
```

Comment:
Is the valgrind complaint is about a missing `else jv_free(home);` here? and i guess it showed up now because shtest on linux etc now runs without HOME set? maybe the window and USERPROFILE test should be separate and not affect the exiting ones?

Reply:
Yes, I had a memory leak.  Valgrind found it because my test didn't correctly restore `HOME` back to its former value (I only `HOME=$HOME_BAK` instead of `export HOME=$HOME_BAK`).  This caused the following test to pass through the `HOME`-not-found codepath.

I've fixed the memory leak and the test.